modules/mono/glue/GodotSharp/GodotSharp/Core/Projection.cs

```diff
@@ -0,0 +1,820 @@
+                i = pvt_i[k];
+                if (i != k)
+                { /* If rows are different */
+                    for (j = 0; j < 4; j++)
+                    {
+                        hold = -proj[k][j];
+                        proj[k, j] = proj[i][j];
+                        proj[i, j] = hold;
+                    }
+                }
+
+                /* "Interchange" columns */
+                j = pvt_j[k];
+                if (j != k)
+                { /* If columns are different */
+                    for (i = 0; i < 4; i++)
+                    {
+                        hold = -proj[i][k];
+                        proj[i, k] = proj[i][j];
+                        proj[i, j] = hold;
+                    }
+                }
+
+                /* Divide column by minus pivot value */
+                for (i = 0; i < 4; i++)
+                {
+                    if (i != k)
+                    {
+                        proj[i, k] /= (-pvt_val);
+                    }
+                }
+
+                /* Reduce the matrix */
+                for (i = 0; i < 4; i++)
+                {
+                    hold = proj[i][k];
+                    for (j = 0; j < 4; j++)
+                    {
+                        if (i != k && j != k)
+                        {
+                            proj[i, j] += hold * proj[k][j];
+                        }
+                    }
+                }
+
+                /* Divide row by pivot */
+                for (j = 0; j < 4; j++)
+                {
+                    if (j != k)
+                    {
+                        proj[k, j] /= pvt_val;
+                    }
+                }
+
+                /* Replace pivot by reciprocal (at last we can touch it). */
+                proj[k, k] = (real_t)1.0 / pvt_val;
+            }
+
+            /* That was most of the work, one final pass of row/column interchange */
+            /* to finish */
+            for (k = 4 - 2; k >= 0; k--)
+            { /* Don't need to work with 1 by 1 corner*/
+                i = pvt_j[k]; /* Rows to swap correspond to pivot COLUMN */
+                if (i != k)
+                { /* If rows are different */
+                    for (j = 0; j < 4; j++)
+                    {
+                        hold = proj[k][j];
+                        proj[k, j] = -proj[i][j];
+                        proj[i, j] = hold;
+                    }
+                }
+
+                j = pvt_i[k]; /* Columns to swap correspond to pivot ROW */
+                if (j != k)
+                { /* If columns are different */
+                    for (i = 0; i < 4; i++)
+                    {
+                        hold = proj[i][k];
+                        proj[i, k] = -proj[i][j];
+                        proj[i, j] = hold;
+                    }
+                }
+            }
+            return proj;
+        }
+
+        public bool IsOrthorgonal()
```

Comment:
There's a typo in the name:

```suggestion
        public bool IsOrthogonal()
```